src/mame/includes/dcheese.h

```diff
@@ -48,27 +53,30 @@ class dcheese_state : public driver_device
 	virtual void device_timer(emu_timer &timer, device_timer_id id, int param, void *ptr) override;
 
 private:
-	required_region_ptr<uint16_t> m_palrom;
-	required_region_ptr<uint8_t> m_gfxrom;
+	required_region_ptr<u16> m_palrom;
+	required_region_ptr<u8> m_gfxrom;
+	required_ioport m_eepromout_io;
+	required_ioport m_2a0002_io;
+	required_ioport m_2a000e_io;
 
 	/* video-related */
-	uint16_t   m_blitter_color[2];
-	uint16_t   m_blitter_xparam[16];
-	uint16_t   m_blitter_yparam[16];
-	uint16_t   m_blitter_vidparam[32];
+	u16   m_blitter_color[2];
+	u16   m_blitter_xparam[16];
+	u16   m_blitter_yparam[16];
+	u16   m_blitter_vidparam[32];
 
 	std::unique_ptr<bitmap_ind16> m_dstbitmap;
 	emu_timer *m_blitter_timer;
 	emu_timer *m_signal_irq_timer;
 
 	/* misc */
-	uint8_t    m_irq_state[5];
-	uint8_t    m_sound_control;
-	uint8_t    m_sound_msb_latch;
+	u8    m_irq_state[5];
+	u8    m_sound_control;
+	u8    m_sound_msb_latch;
 
 	/* devices */
-	required_device<cpu_device> m_maincpu;
-	required_device<cpu_device> m_audiocpu;
+	required_device<m68000_device> m_maincpu;
+	required_device<m6809_device> m_audiocpu;
```

Comment:
Do these actually need to use the derived class?  If not, all it does is increase dependencies for no reason since the CPU headers are now included from the header..